src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -5,15 +5,65 @@ import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import com.fasterxml.jackson.module.kotlin.readValue
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import java.io.File
+import java.io.IOException
 import java.net.URL
 
 class FileSystemYamlReader : YamlReader {
     override fun readToModel(resource: URL): TreeModel {
         val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
         val file = File(resource.toURI())
         val yamlData = file.readText()
+        val treeModel: TreeModel = mapper.readValue(yamlData)
 
-        return mapper.readValue(yamlData)
+        validateTreeModel(treeModel)
+
+        return treeModel
+    }
+
+    private fun validateTreeModel(treeModel: TreeModel) {
+        val titleList = treeModel.nodes.drop(1).map { it.title }
+        val requiresList = treeModel.nodes.flatMap { it.requires.orEmpty() }
+
+        if (titleList != requiresList) {
```

Comment:
```suggestion
        if (!titleList.containsAll(requiresList)) {
```

See: https://github.com/hspsh/tech-tree/pull/6/files#r1493699301

Reply:
Jak się robi takie fajne snippety?

Reply:
CTRL + G albo: 

![image](https://github.com/hspsh/tech-tree/assets/57963469/e8e8eca6-789f-4d6c-bc33-4622d95524b2)

lub też na piechotę:

![image](https://github.com/hspsh/tech-tree/assets/57963469/8b68c359-af5f-477c-a7ae-03bc2332a160)